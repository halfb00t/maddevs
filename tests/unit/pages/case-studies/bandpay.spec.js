import { render } from '@testing-library/vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Bandpay from '@/pages/case-studies/bandpay'
import formBaseProps from '../../__mocks__/formBaseProps'

const localVue = createLocalVue()
localVue.use(Vuex)

const DESCRIPTION = 'FinTech Case Study: Why freelancers don\'t receive payment on time, and how to avoid it? Read Bandpay\'s stories of creating a milestone-based platform to manage the payments process for freelancers.'

const META_DATA = {
  description: DESCRIPTION,
  title: 'Mad Devs Case Study: Bandpay - Payment Management Solution for Freelancers and Small Businesses',
  'og:url': 'https://maddevs.io/case-studies/bandpay/',
  'og:type': 'website',
  'og:site_name': 'Custom Software Development Company',
  'og:title': 'Mad Devs Case Study: Bandpay - Payment Management Solution for Freelancers and Small Businesses',
  'og:description': DESCRIPTION,
  'og:image': 'https://maddevs.io/bandpay-meta-banner.png',
  'og:image:width': '1200',
  'og:image:height': '630',
  'twitter:card': 'summary_large_image',
  'twitter:text:title': 'Mad Devs Case Study: Bandpay - Payment Management Solution for Freelancers and Small Businesses',
  'twitter:description': DESCRIPTION,
  'twitter:image:src': 'https://maddevs.io/bandpay-meta-banner.png',
  'twitter:url': 'https://maddevs.io/case-studies/bandpay/',
  metaTitle: '',
  url: '',
  jsonLd: '{"@context": "https://schema.org", "@type": "WebPage", "name": "Mad Devs Case Study: Bandpay - Payment Management Solution for Freelancers and Small Businesses", "description": "FinTech Case Study: Why freelancers don\'t receive payment on time, and how to avoid it? Read Bandpay\'s stories of creating a milestone-based platform to manage the payments process for freelancers.", "publisher": {"@type": "ProfilePage", "name": "Mad Devs Group Limited"}}',
  image: 'https://maddevs.io/blog.png',
}

const mocks = {
  ...formBaseProps,
  $nuxt: {
    $route: {
      push: jest.fn(),
      path: '/bandpay',
    },
  },
  $getMediaFromS3: () => 'img.jpg',
  $lazyLoad: {
    init: jest.fn(),
  },
}

const stubs = ['NuxtLink', 'Main']

const directives = {
  'lazy-load': () => {},
}

const actions = {
  setHeaderTransparentArea: () => {},
  setHeaderTransparent: () => {},
}

const getters = {
  homePageContent: () => ({ description: DESCRIPTION }),
}

const store = new Vuex.Store({
  actions,
  getters,
})

describe('Bandpay _uid component', () => {
  global.$nuxt = {
    $route: {
      name: null,
    },
  }

  it('should render correctly', () => {
    const { container } = render(Bandpay, {
      localVue,
      store,
      mocks,
      stubs,
      directives,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct work head method', () => {
    const wrapper = shallowMount(Bandpay, {
      localVue,
      store,
      mocks,
      stubs,
      directives,
    })

    const actual = wrapper.vm.$options.head.call(wrapper.vm)

    actual.meta.forEach(meta => {
      expect(META_DATA[meta.name] || META_DATA[meta.property]).toBe(meta.content)
    })
  })
})
