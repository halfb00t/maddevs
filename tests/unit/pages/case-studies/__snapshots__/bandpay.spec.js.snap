// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bandpay _uid component should render correctly 1`] = `
<div>
  <main
    class="case case_bandpay"
  >
    <div
      class="case-bandpay__header-wrapper"
    >
      <section
        class="case_header case_header-bandpay"
        id="transparent-header-area"
      >
        <img
          alt="case video fallback"
          class="case_main-video_fallback"
          data-src="img.jpg"
        />
         
        <!---->
         
        <div
          class="case_header-content"
        >
          <div
            class="case_header-text"
            style="opacity: 1;"
          >
            <div
              class="case_case-study-item"
            >
              
        Case Study
      
            </div>
             
            <h1
              class="case_header-title case_bandpay__header-title"
            >
              
        BandPay: A tool
        
              <br />
              
        to run business
      
            </h1>
             
            <p
              class="case_header-description case_bandpay__header-description"
            >
              
        Mad Devs created a complete B2B payment management solution
        
              <br />
              
        that enables freelancers to promote their business, make
        
              <br />
              
        agreements, and get paid quickly and securely.
      
            </p>
             
          </div>
           
          <img
            alt="bandpay logo"
            class="case_header-logo case_bandpay-logo"
            height="114"
            src="img.jpg"
            style="opacity: 1;"
            width="383"
          />
        </div>
      </section>
    </div>
     
    <main-stub />
     
    <section
      class="case_footer container_full"
    >
      <img
        alt="Guardrails"
        class="case_logotype-guardrails"
        data-src="[object Object]"
        height="113"
        width="230"
      />
       
      <h2
        class="case_title case_title_h2 case_title-guardrails"
      >
        
    Enhancing the GuardRails 
        <br />
        
    solution
  
      </h2>
       
      <nuxtlink-stub
        class="case_next-project-button"
        to="/case-studies/guardrails/"
      >
        
    Read next project
  
      </nuxtlink-stub>
    </section>
  </main>
</div>
`;
